components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1113,6 +1114,35 @@ public IdentityProvider getIdPById(String id, String tenantDomain,
 
         return identityProvider;
     }
+
+    @Override
+    public ExtendedIdentityProvider getIdPByResourceId(String resourceId, String tenantDomain, boolean
+            ignoreFileBasedIdps) throws IdentityProviderManagementException {
+
+        if (StringUtils.isEmpty(resourceId)) {
+            String msg = "Invalid argument: Identity Provider resource ID value is empty";
+            throw new IdentityProviderManagementException(msg);
+        }
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        ExtendedIdentityProvider identityProvider = null;
+
+        identityProvider = dao.getIdPByResourceId(null, resourceId, tenantId, tenantDomain);
+
+        if (!ignoreFileBasedIdps) {
+
+            if (identityProvider == null) {
```

Comment:
Are resourceId based lookup applicable for file-based IDPs?

Reply:
currently resourceId based lookup is not supported for file based IDPs

Reply:
If it is not supported, regardless of the ignoreFileBasedIdps option, we can't send any file-based IDPs

Reply:
removed the ignoreFileBasedIdps logic